### PR TITLE
Updating architecture diagram with "plugins" in a few places.

### PR DESCRIPTION
**Pull request by benvanik**

### CODE REVIEW

**Review by jpienaar — COMMENTED** [review]
I don't know if this actually shows where plugin enables generalization all. It makes it feel like something imported, but it feels closer to PGO feedback, that also feeds into the device codegen. (I feel like runtime has the VM already shown here).

File: docs/website/docs/assets/images/iree_architecture_dark.svg
Comment (ScottTodd) [inline docs/website/docs/assets/images/iree_architecture_dark.svg:1]: Plugin in the top left of this file looks hard to read - maybe change the colors to match the other items?

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/website/docs/assets/images/iree_architecture_dark.svg
Comment (ScottTodd) [reply docs/website/docs/assets/images/iree_architecture_dark.svg:1]: Preview here: https://github.com/openxla/iree/tree/benvanik-arch-svg#architecture-overview

(GitHub diff makes it look hard to read, when it's actually a little _too_ easy to read with the higher contrast than the other items)

image

Comment (ScottTodd) [reply docs/website/docs/assets/images/iree_architecture_dark.svg:1]: Sent https://github.com/openxla/iree/pull/12856 with some tweaks ("Plugin" font color, "IREE Importers" background/stroke color)

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**